apis/clouddms/v1alpha1/migrationjob_types.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common/parent"
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var CloudDMSMigrationJobGVK = GroupVersion.WithKind("CloudDMSMigrationJob")
+
+// CloudDMSMigrationJobSpec defines the desired state of CloudDMSMigrationJob
+// +kcc:spec:proto=google.cloud.clouddms.v1.MigrationJob
+type CloudDMSMigrationJobSpec struct {
+	// Required. Defines the parent path of the resource.
+	*parent.ProjectAndLocationRef `json:",inline"`
+
+	// The resource labels for migration job to use to annotate any related
+	//  underlying resources such as Compute Engine VMs. An object containing a
+	//  list of "key": "value" pairs.
+	//
+	//  Example: `{ "name": "wrench", "mass": "1.3kg", "count": "3" }`.
+	// +kcc:proto:field=google.cloud.clouddms.v1.MigrationJob.labels
+	// Labels map[string]string `json:"labels,omitempty"`
+
+	// The migration job display name.
+	// +kcc:proto:field=google.cloud.clouddms.v1.MigrationJob.display_name
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Required. The migration job type.
+	// +kcc:proto:field=google.cloud.clouddms.v1.MigrationJob.type
+	Type *string `json:"type,omitempty"`
+
+	// The path to the dump file in Google Cloud Storage,
+	//  in the format: (gs://[BUCKET_NAME]/[OBJECT_NAME]).
+	//  This field and the "dump_flags" field are mutually exclusive.
+	// +kcc:proto:field=google.cloud.clouddms.v1.MigrationJob.dump_path
+	DumpPath *string `json:"dumpPath,omitempty"`
+
+	// The initial dump flags.
+	//  This field and the "dump_path" field are mutually exclusive.
+	// +kcc:proto:field=google.cloud.clouddms.v1.MigrationJob.dump_flags
+	DumpFlags *MigrationJob_DumpFlags `json:"dumpFlags,omitempty"`
+
+	// Required. The resource name (URI) of the source connection profile.
```

Comment:
Nit: I tend to change the comments that refer to URI or name, to avoid "type" information.